templates/contribute/style.md

```diff
@@ -565,3 +565,29 @@ Documentation strings for declarations are delimited with `/-- -/`.
 
 See our [documentation requirements](doc.html) for more suggestions
 and examples.
+
+### Deprecation
+
+Deleting, renaming, or changing declarations can cause downstreams projects
+that rely on these definitions to fail to compile. Any publicly exposed theorems and
+definitions that are being removed should be gracefully transitioned by keeping the
+old theorem statement with a `@[deprecated]` attribute, to warn downstream projects to
+transition away from deprecated definitions before they are deleted. Renamed theorems
+can use a deprecated `alias` to the new name, while in cases where the theorem is not
+being replaced, the deprecated theorem definition can be marked with an attribute, like so:
+
+```lean4
+theorem new_name : ... := ...
+@[deprecated new_name (since := "YYYY-MM-DD")] alias old_name := new_name
+
+@[deprecated "This theorem is deprecated in favor of using ... with ..." (since := "YYYY-MM-DD")]
+theorem example_thm ...
+```
+
+The `@[deprecated]` attribute should always include the deprecation date,
```

Comment:
If I'm reading https://github.com/leanprover/lean4/pull/6112 correctly, the attribute already enforces this. So perhaps
```suggestion
The `@[deprecated]` attribute requires the deprecation date,